src/protocol/sort/compose.rs

```diff
@@ -102,20 +104,20 @@ mod tests {
             let mut rho_composed = rho_u128.clone();
             apply_inv(&mut Permutation::oneline(sigma.clone()), &mut rho_composed);
 
-            let mut sigma_shares = generate_shares::<Fp31>(sigma_u128);
+            let sigma_shares = generate_shares::<Fp31>(sigma_u128);
             let mut rho_shares = generate_shares::<Fp31>(rho_u128);
             let world: TestWorld = make_world(QueryId);
             let [ctx0, ctx1, ctx2] = make_contexts(&world);
 
-            let mut compose0 = Compose::new(&mut sigma_shares.0, &mut rho_shares.0);
-            let mut compose1 = Compose::new(&mut sigma_shares.1, &mut rho_shares.1);
-            let mut compose2 = Compose::new(&mut sigma_shares.2, &mut rho_shares.2);
+            let mut compose0 = Compose::new(sigma_shares.0, rho_shares.0);
+            let mut compose1 = Compose::new(sigma_shares.1, rho_shares.1);
+            let mut compose2 = Compose::new(sigma_shares.2, rho_shares.2);
 
             let h0_future = compose0.execute(ctx0);
             let h1_future = compose1.execute(ctx1);
             let h2_future = compose2.execute(ctx2);
 
-            try_join!(h0_future, h1_future, h2_future)?;
+            rho_shares = try_join!(h0_future, h1_future, h2_future)?;
```

Comment:
IMO you shouldn't reuse the same variable unless there's a specific reason to (for instance, if you want to modify a borrowed value). in this case, would it be more clear if you did something like
```suggestion
            let composed_rho_shares = try_join!(h0_future, h1_future, h2_future)?;
```